specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/costmanagement.benefits.json

```diff
@@ -0,0 +1,1002 @@
+          "readOnly": true
+        },
+        "averageUtilizationPercentage": {
+          "description": "Estimated average utilization percentage at this commitment.",
+          "type": "number",
+          "format": "decimal",
+          "readOnly": true
+        },
+        "wastageCost": {
+          "description": "Wastage cost at this commitment.",
+          "type": "number",
+          "format": "decimal",
+          "readOnly": true
+        }
+      }
+    },
+    "BenefitPlanKind": {
+      "type": "string",
+      "description": "Kind of the benefit plan.",
+      "enum": [
+        "Reservation",
+        "SavingsPlan"
+      ],
+      "x-ms-enum": {
+        "name": "benefitPlanKind",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Reservation",
+            "description": "Benefit Plan is for a Reservation.",
+            "name": "Reservation"
+          },
+          {
+            "value": "SavingsPlan",
+            "description": "Benefit Plan is for a SavingsPlan.",
+            "name": "SavingsPlan"
+          }
+        ]
+      }
+    },
+    "RecommendationScope": {
+      "type": "string",
+      "description": "Kind of the recommendation scope.",
+      "enum": [
+        "Single",
+        "Shared"
+      ],
+      "x-ms-enum": {
+        "name": "scope",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Single",
+            "description": "Single scope recommendation.",
+            "name": "Single"
+          },
+          {
+            "value": "Shared",
+            "description": "Shared scope recommendation.",
+            "name": "Shared"
+          }
+        ]
+      }
+    },
+    "Grain": {
+      "type": "string",
+      "description": "Grain which corresponds to value.",
+      "enum": [
+        "Hourly",
+        "Daily",
+        "Monthly"
+      ],
+      "x-ms-enum": {
+        "name": "grain",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "Hourly",
+            "description": "Hourly grain corresponds to value per hour.",
+            "name": "Hourly"
+          },
+          {
+            "value": "Daily",
+            "description": "Hourly grain corresponds to value per day.",
+            "name": "Daily"
+          },
+          {
+            "value": "Monthly",
+            "description": "Hourly grain corresponds to value per month.",
+            "name": "Monthly"
+          }
+        ]
+      }
+    },
+    "BenefitResource": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Resource ID.",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "description": "Resource name.",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "Resource type.",
+          "readOnly": true
+        },
```

Comment:
These should really all be inherited from the common-types definitions like ProxyResource somehow. Sorry if I give the same feedback multiple times but too many PRs.

Reply:
Done.